lua/nvim-tree.lua

```diff
@@ -342,6 +342,18 @@ local function setup_autocommands(opts)
   if opts.reload_on_bufenter then
     create_nvim_tree_autocmd("BufEnter", { pattern = "NvimTree_*", callback = reloaders.reload_explorer })
   end
+
+  if opts.view.centralize_selection then
+    create_nvim_tree_autocmd("BufEnter", {
+      pattern = "NvimTree_*",
+      callback = function()
+        vim.schedule(function()
+          local keys = api.nvim_replace_termcodes("zz", true, false, true)
+          api.nvim_feedkeys(keys, "n", true)
```

Comment:
Does not obey `hijack_cursor = true`.

Putting this autocmd before `CursorMoved` does not fix this.

Please try changing this autocmd to be nested to see if you can make hijack_cursor work. If it's not possible I am happy to let it go.

Reply:
I fixed it for now, but I do think that the problem originates from event used for `hijack_cursor`, example: try `:NvimTreeToggle` while having `hijack_cursor` to be set.  I don't think this would've been a problem, if the initial "legal" position is set when entering.

That said, I'm not sure it's worth the effort or the complexity to try to cover all the edge-cases that would originate from such a change.